airflow/utils/state.py

```diff
@@ -122,3 +122,10 @@ def unfinished(cls):
             cls.UP_FOR_RETRY,
             cls.UP_FOR_RESCHEDULE
         ]
+
+    @classmethod
+    def unsuccessful(cls):
+        """
+        A list of states indicating that a task completed unsuccessfully.
+        """
+        return [cls.FAILED, cls.UP_FOR_RETRY, cls.UP_FOR_RESCHEDULE]
```

Comment:
Not sure reschedule should be in this list -- it's used by sensors to retry but not tie up a slot. But this is basically still "running" otherwise.

Reply:
UP_FOR_RESCHEDULE is actually added per your comment 24 days ago.
"Up for reschedule too."
So please let me know what's your thought so I can make a final change. 

IMO,  UP_FOR_RESCHEDULE should not be treated as a unsuccessful state as you describe.

Reply:
:man_facepalming:

Reply:
@lucianoviola That link doesn't take me to any discussion (anymore)?

Reply:
@ashb Maybe it's because it's already resolved. Here is a screenshot
![image](https://user-images.githubusercontent.com/1295785/72370332-dee94a00-36b6-11ea-94c6-bafb5da91c90.png)

Reply:
Past me was right, and this is all good here.